### PR TITLE
bug(locators): excapt query in byExactBinding

### LINKED ISSUE #1918: exactBinding fails to locate bindings which contain certain characters
bindings which use characters that have special meaning in regular expressions will fail to work in By.exactBinding - the binding needs to be escaped before matching. It is deceiving because of the way you have the argument described:

```javascript
@param {string} binding The binding, e.g. {{cat.name}}.
```

Example: 
```html
<span>{{ vehicle.garagingTerritory.name() }}</span>
```

passing in 'vehicle.garagingTerritory.name()' to By.exactBinding will not match the element. You must at least escape the parenthesis. I would suggest escaping all of the input automatically, or adjusting the documentation to describe when the user needs to escape the input themselves.

Thanks

### PR DESCRIPTION
**Pull request by sjelin**
See http://stackoverflow.com/q/3561711

Closes #1918

### CODE REVIEW

**Comment by bootstraponline** [comment]
:+1:

File: lib/clientsidescripts.js [hunk 1]
```diff
@@ -72,7 +72,10 @@ functions.findBindings = function(binding, exactMatch, using, rootSelector) {
     if (dataBinding) {
       var bindingName = dataBinding.exp || dataBinding[0].exp || dataBinding;
       if (exactMatch) {
-        var matcher = new RegExp('({|\\s|^|\\|)' + binding + '(}|\\s|$|\\|)');
+        var matcher = new RegExp('({|\\s|^|\\|)' +
+            // See http://stackoverflow.com/q/3561711
```
Comment (juliemr) [inline lib/clientsidescripts.js]: Single line comments inside client side scripts can screw things up - see the note at the top of this file. Make this `/* */` style.

**Comment by juliemr** [comment]
One comment, then LGTM.

**Comment by sjelin** [comment]
Merged in at 3cded9b

**Pull request closed without merging**